dbcon/joblist/rowestimator.cpp

```diff
@@ -140,12 +140,13 @@ uint64_t RowEstimator::adjustValue(const execplan::CalpontSystemCatalog::ColType
 
 // Estimates the number of distinct values given a min/max range.  When the range has not been set,
 // rules from the requirements are used based on the column type.
+template<typename T>
 uint32_t RowEstimator::estimateDistinctValues(const execplan::CalpontSystemCatalog::ColType& ct,
-        const uint64_t& min,
-        const uint64_t& max,
+        const T& min,
+        const T& max,
         const char cpStatus)
 {
-    uint64_t ret = 10;
+    T ret = 10;
```

Comment:
It seems like an automated change has been applied here. 
We return uint32_t from this f() into uint32_t variable so ret should be uint32_t not even uint64_t. Given the calculations this should be safe.

Reply:
This is intentional. The code at the end of this function is this:

```
    else
    {
        ret = max - min + 1;
    }

    if (ret > (T) fRowsPerExtent)
    {
        ret = fRowsPerExtent;
    }

    return ret;
```

So the type of ret has to be same as max/min, since we perform arithmetic on these values. Subsequently, `if (ret > (T) fRowsPerExtent)` ensures the maximum value returned fits in an uint32_t.